src/preprocessing/passes/bv_to_int.cpp

```diff
@@ -269,560 +242,542 @@ Node BVToInt::bvToInt(Node n)
 {
   n = eliminationPass(n);
   n = makeBinary(n);
-  vector<Node> toVisit;
-  toVisit.push_back(n);
   uint64_t granularity = options::solveBVAsInt();
 
-  while (!toVisit.empty())
+  for (TNode current : NodeDfsIterable(n).inPostorder().skipIf(
+           [this](TNode nn) { return d_bvToIntCache.count(nn) > 0; }))
   {
-    Node current = toVisit.back();
     uint64_t currentNumChildren = current.getNumChildren();
-    if (d_bvToIntCache.find(current) == d_bvToIntCache.end())
+    if (d_bvToIntCache.find(current) != d_bvToIntCache.end())
```

Comment:
Is this condition ever `false` given the function given to `skipIf`?

Reply:
Yes, this shouldn't be needed anymore since skipIf takes care of that, @alex-ozdemir?

Reply:
`skipIf` does take care of this. I'll change this accordingly.

Note though, that the `skipIf` PR hasn't landed yet, and is waiting on your review @mpreiner.